### PR TITLE
[BugFix] Fix load with incomplete sort key

### PR DESCRIPTION
**Pull request by TszKitLo40**
Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/delta_writer.cpp|1|1|100.00%|[]|

File: be/test/storage/lake/partial_update_test.cpp [hunk 1]
```diff
@@ -604,4 +604,149 @@ TEST_P(LakePartialUpdateTest, test_write_with_index_reload) {
...
+    }
+
+    void TearDown() override {
+        // check primary index cache's ref
+        EXPECT_TRUE(_update_mgr->TEST_check_primary_index_cache_ref(_tablet_metadata->id(), 1));
+        ExecEnv::GetInstance()->vacuum_thread_pool()->wait();
+        // check trash files already removed
+        for (const auto& file : _trash_files) {
+            EXPECT_FALSE(fs::path_exist(file));
+        }
```
Comment (sduzh) [inline be/test/storage/lake/partial_update_test.cpp:683]:
```suggestion
```

File: be/test/storage/lake/partial_update_test.cpp [hunk 2]
```diff
@@ -604,4 +604,149 @@ TEST_P(LakePartialUpdateTest, test_write_with_index_reload) {
...
+    }
+
+protected:
+    constexpr static const char* const kTestDirectory = "test_lake_partial_update";
+    constexpr static const int kChunkSize = 12;
+
+    std::unique_ptr<TabletMetadata> _tablet_metadata;
+    std::shared_ptr<TabletSchema> _tablet_schema;
+    std::shared_ptr<Schema> _schema;
+    std::shared_ptr<Schema> _partial_schema;
```
Comment (sduzh) [inline be/test/storage/lake/partial_update_test.cpp:715]: unused variable

File: be/test/storage/lake/partial_update_test.cpp [hunk 3]
```diff
@@ -604,4 +604,149 @@ TEST_P(LakePartialUpdateTest, test_write_with_index_reload) {
...
+        return Chunk({c0, c1}, _slot_cid_map);
+    }
+
+protected:
+    constexpr static const char* const kTestDirectory = "test_lake_partial_update";
+    constexpr static const int kChunkSize = 12;
+
+    std::unique_ptr<TabletMetadata> _tablet_metadata;
+    std::shared_ptr<TabletSchema> _tablet_schema;
+    std::shared_ptr<Schema> _schema;
```
Comment (sduzh) [inline be/test/storage/lake/partial_update_test.cpp:714]: unused

File: be/test/storage/lake/partial_update_test.cpp [hunk 4]
```diff
@@ -604,4 +604,149 @@ TEST_P(LakePartialUpdateTest, test_write_with_index_reload) {
...
+    std::shared_ptr<Schema> _partial_schema;
+    int64_t _partition_id = 4561;
+    std::vector<std::string> _trash_files;
```
Comment (sduzh) [inline be/test/storage/lake/partial_update_test.cpp:717]: unused

**Review by sduzh — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/delta_writer.cpp|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/delta_writer.cpp|1|1|100.00%|[]|

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**